### PR TITLE
Discover discs

### LINKED ISSUE #36: Agent: discover disks

### PR DESCRIPTION
**Pull request by wkucjas**
Closes #36

### CODE REVIEW

File: src/agent/Disc.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <string>
+
+class Disc
+{
+public:
+    Disc();
+    Disc(std::string _caption, std::string _deviceId, std::string _model, int _partitions, int _size);
```
Comment (Kicer86) [inline src/agent/Disc.h]: const &

File: src/agent/IDiscCollector.h [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+#include <vector>
+#include <string>
+
+class IDiscCollector
+{
+public:
+
+    virtual ~IDiscCollector() = default;
+    virtual std::vector<std::string> GetDiscsList();
```
Comment (Kicer86) [inline src/agent/IDiscCollector.h]: what will be returned in this vector?

**Review by Kicer86 — CHANGES_REQUESTED** [review]

File: src/agent/IDiscCollector.h (hunk 2 above)
Comment (wkucjas) [reply src/agent/IDiscCollector.h]: Nobody know now. It's work in progress :)

File: src/agent/windows/WMICommunication.h [hunk 3]
```diff
@@ -4,26 +4,39 @@
 #include <memory>
 #include <Wbemidl.h>
 #include "common/SmartData.h"
+#include "../Disk.h"
 
 
 
 class WMICommunication
```
Comment (Kicer86) [inline src/agent/windows/WMICommunication.h:11]: Just to consider: one base class for common stuff which can be inherited and used by specialized classes

**Review by Kicer86 — APPROVED** [review]

File: src/agent/windows/WMICommunication.h (hunk 3 above)
Comment (wkucjas) [reply src/agent/windows/WMICommunication.h:11]: Issue #39 added. Will be addressed in future.

**Pull request merged**